keps/sig-network/20180612-ipv4-ipv6-dual-stack.md

```diff
@@ -526,302 +517,382 @@ dualstack cluster can have this flag configured as
+>  `spec.ipFamilies : ["IPv4"]` and `spec.clutserIPs: ["<ipv6>" ]` is
+>  considered invalid because `ipFamilies[0]` does not match `ClusterIPs[0]`.
 
 The rules that govern these fields are described in the below sections
 
 ##### Service Object Mutability Rules
 
-The existing mutability rules will remain as is. Primarily services can not
-change their `ClusterIP` (as we move to dual stack, this is considered as
-`Service Primary ClusterIP`), yet services can change types (e.g. from
-`ClusterIP` to `ExternalName`). The following are additional immutability rules
+The existing mutability rules will remain as is. Services can not
+change their `clusterIP` (aka `clusterIPs[0]`) - this is considered as
+`Service Primary ClusterIP`, Services can still change `type`s as before (e.g.
+from `ClusterIP` to `ExternalName`). The following are additional immutability
+rules
 
 1. Services can change from `Single Stack` to `Dual Stack (required)` or `Dual
    Stack (optional)` as described below.
 2. Services can also change `Dual Stack` to `Single Stack`.
-3. Services `can not` change the primary service IP `spec.clusterIP` or
+3. Services can not change the primary service IP `spec.clusterIP` or
    `spec.clusterIPs[0]` because this will break existing mutability rules. This
    also apply even changing a service from `Single Stack` to `Dual Stack` or
    the other way around.
 
 ##### Creating a New Single Stack Service
 
-User who want to create a new single stack Service can create it using one of
-the following methods:
-1. Set `spec.preferDualStack: nil` and `spec.ipFamilies: nil` fields.
-   api-server will default `spec.preferDualStack: false` and `spec.ipFamilies:
-   ['default IP family of the cluster']`.
-2. Set `spec.preferDualStack: false` api-server will default `spec.ipFamilies:
-   ['cluster default IP address family']`.
-3. Set `spec.preferDualStack: nil` and `spec.ipFamilies: ['some IP family']`
-   api-server will default `spec.preferDualStack: false`. api-server will fail
-   the request if the IP family is not configured on the cluster.
-4. Set `spec.preferDualStack: false` and `spec.ipFamilies: ['some IP family']`
-   api-server will fail the request if the IP family is not configured on the
-   cluster.
-5. Set `spec.clusterIPs: ['some IP']` api-server will default
-   `spec.preferDualStack:false` and `spec.ipFamilies: ['IP family of
-   spec.clusterIP[0]']`. api-server will fail the request if the IP family is
-   not configured on the cluster or the IP is out of CIDR range or IP is
-   already allocated to a different service.
+Users who want to create a new single stack Service can create it using one of
+the following methods (in increasing specificity):
+1. Do not set `spec.preferDualStack`, `spec.ipFamilies`, or `spec.clusterIPs`
+   fields. The apiserver will default `spec.preferDualStack` to `false`, will
+   assign `spec.ipFamilies` to the cluster's default IP family, and will
+   allocate a `clusterIP` of that same family.
```

Comment:
Possibly worth pointing out that this case applies to all legacy Service definitions

Reply:
Will add some words a bit farther away - before all of this.